docs/troubleshooting/kubernetes.rst

```diff
@@ -1,13 +1,223 @@
+
+-----------------------------------------
+
+Why didn't the |kctlr| create any objects on the BIG-IP system?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here are a few things to check:
+
+Does the namespace of the Kubernetes resource match the namespace(s) you set the Controller to watch?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+   :caption: Excerpt from a sample Deployment
+
+   apiVersion: extensions/v1beta1
+   kind: Deployment
+   metadata:
+     name: k8s-bigip-ctlr-deployment
+     namespace: kube-system
+   ...
+             args: [
+               "--bigip-username=$(BIGIP_USERNAME)",
+               "--bigip-password=$(BIGIP_PASSWORD)",
+               "--bigip-url=10.190.24.171",
+               "--bigip-partition=kubernetes",
+               "--namespace=<my-namespace>",
+               ]
+   ...
+
+
+Are the Service name and port provided in your virtual server ConfigMap correct?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+   :caption: Sample Kubernetes Service
+   :emphasize-lines: 4, 11
+
+   kind: Service
+   apiVersion: v1
+   metadata:
+     name: hello
+   spec:
+     selector:
+       app: hello
+       tier: backend
+     ports:
+     - protocol: TCP
+       port: 80
+       targetPort: http
+
+*Source:* `Connect a Front End to a Back End Using a Service <https://kubernetes.io/docs/tasks/access-application-cluster/connecting-frontend-backend/#creating-the-backend-service-object>`_
+
+.. code-block:: yaml
+   :caption: Excerpt from a sample virtual server ConfigMap
+   :emphasize-lines: 10-11
+
+   kind: ConfigMap
+   apiVersion: v1
+   ...
+   data:
+    schema: "f5schemadb://bigip-virtual-server_v0.1.4.json"
+     data: |
+       {
+         "virtualServer": {
+           "backend": {
+             "servicePort": 80,
+             "serviceName": "hello",
+           },
+      ...
+       }
+
+Does the `service type`_ provided in the BIG-IP Controller Deployment match the service type defined for the Service?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- The default service type used by Kubernetes is ``clusterIP``.
+- The default service type used by the |kctlr| is ``nodeport``.
+
+**If you didn't specify a service type** in the Service definition or the |kctlr| Deployment, you probably have a service type mismatch. See :ref:`Nodeport mode vs Cluster mode <kctlr modes>` for more information about each service type and recommended use.
+
+
+Did you provide valid JSON?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The |kctlr| can only parse valid JSON. Run your desired configurations through a JSON linter before use to avoid potential object creation errors.
```

Comment:
We should mention here that this is specifically referring to the ConfigMap `data.data` field. The API will catch invalid YAML/JSON throughout the resources but this field is passed through the API server as a simple string. This field `data.data` is a string that must be formatted as valid JSON for the controller to parse.